partiql-types/src/main/kotlin/org/partiql/types/StaticType.kt

```diff
@@ -245,18 +173,11 @@ sealed class StaticType {
 // TODO: Remove `NULL` from here. This affects inference as operations (especially NAry) can produce
 //  `NULL` or `MISSING` depending on a null propagation or an incorrect argument.
 data class AnyType(override val metas: Map<String, Any> = mapOf()) : StaticType() {
-    // AnyType encompasses all PartiQL types (including Null type)
-    override fun isInstance(value: ExprValue): Boolean = true
-    override fun isComparableTo(other: StaticType): Boolean = true
-
-    override val typeDomain: Set<ExprValueType>
-        get() = enumValues<ExprValueType>().toSet()
-
     /**
      * Converts this into an [AnyOfType] representation. This method is helpful in inference when
      * it wants to iterate over all possible types of an expression.
      */
-    fun toAnyOfType() = AnyOfType(typeDomain.map { fromExprValueType(it) }.toSet())
+    fun toAnyOfType() = AnyOfType(ALL_TYPES.toSet())
```

Comment:
This changes the semantics significantly. 
Perhaps this method was meant to be moved to StaticTypeUtils, without the implementation change?